packages/block-library/src/image/edit.js

```diff
@@ -293,8 +303,13 @@ class ImageEdit extends Component {
 		this.props.setAttributes( { ...extraUpdatedAttributes, align: nextAlign } );
 	}
 
-	updateImageURL( url ) {
-		this.props.setAttributes( { url, width: undefined, height: undefined } );
+	updateImageURL( url, data ) {
+		this.props.setAttributes( {
+			url,
```

Comment:
As discussed during the pairing session earlier today, how about we use `slug` from media sizes object as a value here? The other possible approach is to detect `slug` based on the media sizes object.